@stellar/design-system/src/components/PieProgress/index.tsx

```diff
@@ -0,0 +1,42 @@
+import "./styles.scss";
+
+interface PieProgressProps {
+  total: number;
+  passed: number;
+  failed: number;
+  customSize?: string;
+  customBackgroundColor?: string;
+}
+
+export const PieProgress = ({
+  total,
+  passed,
+  failed,
+  customSize,
+  customBackgroundColor,
+}: PieProgressProps) => {
+  const passedPercentage = Math.round((passed / total) * 100);
+  const failedPercentage = Math.round((failed / total) * 100);
```

Comment:
do you think it's worth throwing an error if passed + failed > total to prevent any unexpected behavior?

Reply:
I thought about that but checked that it doesn't break and left as is. It would just show passed over failed. It would be mostly to alert developers, right? Would we want the user to see the error if this happens in the production? 🤔

Reply:
Yeah, I was thinking more about a dev who accidentally put in the wrong values. Maybe just a console.error would be enough, but up to you!

Reply:
Added check only in development. We don't have more elegant error handling in SDS yet, but at least it will give some feedback to developers.